packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -466,6 +469,26 @@ export default function useAutocomplete(props) {
       return;
     }
 
+    if (
+      highlightedIndexRef.current !== -1 &&
+      previousProps.filteredOptions &&
+      previousProps.filteredOptions.length !== filteredOptions.length &&
+      (multiple
+        ? previousProps.value.every((val, i) => getOptionLabel(value[i]) === getOptionLabel(val))
+        : getOptionLabel(previousProps.value ?? '') === getOptionLabel(value ?? ''))
+    ) {
```

Comment:
This condition is complex and takes a while to parse and understand it. Could you please add a comment and/or extract it to a function with a meaningful name to help others understand what this code is responsible for?

Reply:
done, commit for requested change: https://github.com/mui/material-ui/pull/35735/commits/c75f72ef185457321e0acc3aa3f331855bfe6487